moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -507,7 +506,9 @@ export function useMapDrawTools(
         ...featureCollection.features
           .filter(
             // Keep the features that are not equal to featureIdToDelete
-            feature => get(feature, featureIdGetPath) !== featureIdToDelete
+            feature =>
+              !featureIdToDelete ||
+              get(feature, featureIdGetPath) !== featureIdToDelete
```

Comment:
In the process of simplifying the data handling in `ProjectComponentEdit.js`, i introduced a bug that would cause selected features to be dropped if the user used a mix of drawn and selected features. This big ensures that only drawn features are included in the state handling removes them from component's feature collection when they are deleted through the draw interface.